Jellyfin.Data/Entities/Book.cs

```diff
@@ -21,64 +9,67 @@
+        /// </summary>
+        public static Book CreateBookUnsafe()
+        {
+            return new Book();
+        }
 
-      /// <summary>
-      /// Public constructor with required data
-      /// </summary>
-      /// <param name="urlid">This is whats gets displayed in the Urls and API requests. This could also be a string.</param>
-      public Book(Guid urlid, DateTime dateadded)
-      {
-         this.UrlId = urlid;
+        /// <summary>
+        /// Public constructor with required data
+        /// </summary>
+        /// <param name="urlid">This is whats gets displayed in the Urls and API requests. This could also be a string.</param>
+        public Book(Guid urlid, DateTime dateadded)
+        {
+            this.UrlId = urlid;
 
-         this.BookMetadata = new System.Collections.Generic.HashSet<global::Jellyfin.Data.Entities.BookMetadata>();
-         this.Releases = new System.Collections.Generic.HashSet<global::Jellyfin.Data.Entities.Release>();
+            this.BookMetadata = new HashSet<BookMetadata>();
+            this.Releases = new HashSet<Release>();
 
-         Init();
-      }
+            Init();
+        }
 
-      /// <summary>
-      /// Static create function (for use in LINQ queries, etc.)
-      /// </summary>
-      /// <param name="urlid">This is whats gets displayed in the Urls and API requests. This could also be a string.</param>
-      public static Book Create(Guid urlid, DateTime dateadded)
-      {
-         return new Book(urlid, dateadded);
-      }
+        /// <summary>
+        /// Static create function (for use in LINQ queries, etc.)
+        /// </summary>
+        /// <param name="urlid">This is whats gets displayed in the Urls and API requests. This could also be a string.</param>
+        public static Book Create(Guid urlid, DateTime dateadded)
+        {
+            return new Book(urlid, dateadded);
+        }
 
-      /*************************************************************************
-       * Properties
-       *************************************************************************/
+        /*************************************************************************
+         * Properties
+         *************************************************************************/
 
-      /*************************************************************************
-       * Navigation properties
-       *************************************************************************/
+        /*************************************************************************
+         * Navigation properties
+         *************************************************************************/
 
-      public virtual ICollection<global::Jellyfin.Data.Entities.BookMetadata> BookMetadata { get; protected set; }
+        [ForeignKey("BookMetadata_BookMetadata_Id")]
+        public virtual ICollection<BookMetadata> BookMetadata { get; protected set; }
```

Comment:
Something seems wrong about how the relationships are configured. Normally you don't need to use the `[ForeignKey]` attribute at all if the models are set up correctly.

For example, in this case I would expect the `BookMetadata` class to have `BookId` and `Book` navigation properties, in which case the `[ForeignKey]` attribute is not needed because the relationship is specified by convention. Maybe there were issues getting this to work because the navigation properties on the other side of the relationship are missing?

Reply:
cc @EraYaN

Reply:
This is to specify the name of the foreign key, even if there is no navigation property. But if you want to add all of them, feel free. This is just a direct port of the info the generator put in, and it works.

Reply:
Fair enough, can be fixed later if necessary. I wonder why the generator didn't add the navigation properties to be begin with though :/